src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -291,7 +291,7 @@ public void usage(final PrintStream stream, final boolean printCommon) {
             if (optArgs != null && !optArgs.isEmpty()) {
                 stream.println("\nOptional Arguments:\n");
                 optArgs.sort(ArgumentDefinition.sortByLongName);
-                optArgs.stream().forEach(argumentDefinition -> printArgumentUsage(stream, argumentDefinition));
+                optArgs.stream().filter(argDef -> !argDef.isHidden).forEach(argumentDefinition -> printArgumentUsage(stream, argumentDefinition));
```

Comment:
Also, I think the filter should be moved up to the beginning of this method, where it will filter all arguments right from the start.